pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,446 @@
+	usage += "  {{ProgramName}} adm log-verbosity --reset --virt-handler=7\n"
+	usage += "  # set all components to 3 besides virt-handler which is 7\n"
+	usage += "  {{ProgramName}} adm log-verbosity --all=3 --virt-handler=7\n"
+
+	return usage
+}
+
+// VirtComponent to component name
+// also used by the test file
+func GetComponentNameByVirtComponent(component virtComponent) string {
+	var virtComponentToComponentName = map[virtComponent]string{
+		VirtAPI:        "virt-api",
+		VirtController: "virt-controller",
+		VirtHandler:    "virt-handler",
+		VirtLauncher:   "virt-launcher",
+		VirtOperator:   "virt-operator",
+		All:            "all",
+	}
+	return virtComponentToComponentName[component]
+}
+
+// virtComponent to JSON name
+func getJSONNameByVirtComponent(component virtComponent) string {
+	var virtComponentToJSONName = map[virtComponent]string{
+		VirtAPI:        "virtAPI",
+		VirtController: "virtController",
+		VirtHandler:    "virtHandler",
+		VirtLauncher:   "virtLauncher",
+		VirtOperator:   "virtOperator",
+		All:            "all",
+	}
+	return virtComponentToJSONName[component]
+}
+
+// component name to JSON name
+func getJSONNameByComponentName(componentName string) string {
+	var componentNameToJSONName = map[string]string{
+		"virt-api":        "virtAPI",
+		"virt-controller": "virtController",
+		"virt-handler":    "virtHandler",
+		"virt-launcher":   "virtLauncher",
+		"virt-operator":   "virtOperator",
+		"all":             "all",
+	}
+	return componentNameToJSONName[componentName]
+}
+
+func detectInstallNamespaceAndName(virtClient kubecli.KubevirtClient) (namespace, name string, err error) {
+	// listing KubeVirt CRs in all namespaces and see where it is installed
+	kvs, err := virtClient.KubeVirt("").List(&k8smetav1.ListOptions{})
+	if err != nil {
+		return "", "", fmt.Errorf("could not list KubeVirt CRs in all namespaces: %v", err)
+	}
+	if len(kvs.Items) == 0 {
+		return "", "", errors.New("could not detect a KubeVirt installation")
+	}
+	if len(kvs.Items) > 1 {
+		return "", "", errors.New("invalid kubevirt installation, more than one KubeVirt resource found")
+	}
+	namespace = kvs.Items[0].Namespace
+	name = kvs.Items[0].Name
+	return
+}
+
+func hasVerbosityInKV(kv *v1.KubeVirt) (map[string]uint, error) {
+	verbosityMap := map[string]uint{} // key: component name, value: verbosity
+	// check the logVerbosity field in the KubeVirt CR
+	lvJSON, err := json.Marshal(kv.Spec.Configuration.DeveloperConfiguration.LogVerbosity)
+	if err != nil {
+		return nil, err
+	}
+	// if there is a logVerbosity field, input the JSON data to the verbosityMap
+	if err := json.Unmarshal(lvJSON, &verbosityMap); err != nil {
+		return nil, err
+	}
+	return verbosityMap, nil
+}
+
+func createOutputLines(verbosityVal map[string]uint, options map[string]uint) []string {
+	var lines []string
+	allIsSet := false
+	if _, exist := options["all"]; exist {
+		allIsSet = true
+	}
```

Comment:
I think this can be simplified

```diff
-       allIsSet := false
-       if _, exist := options["all"]; exist {
-               allIsSet = true
-       }
+       _, allIsSet := options["all"]
```

Reply:
Great. Thanks.